operator/slotticker/slotticker.go

```diff
@@ -81,10 +84,15 @@ func (s *slotTicker) Next() <-chan time.Time {
 		default:
 		}
 	}
-	slotNumber := uint64(timeSinceGenesis / s.slotDuration)
-	nextSlotStartTime := s.genesisTime.Add(time.Duration(slotNumber+1) * s.slotDuration)
+	currentSlot := phase0.Slot(timeSinceGenesis / s.slotDuration)
+	if currentSlot <= s.slot {
+		// We've already ticked for this slot, so we need to wait for the next one.
+		currentSlot = s.slot
+		s.logger.Warn("double tick", zap.Uint64("slot", uint64(currentSlot)))
+	}
+	nextSlotStartTime := s.genesisTime.Add(time.Duration(currentSlot+1) * s.slotDuration)
```

Comment:
@MatusKysel , we've seen this happen in e2e, where <-Next(), Slot(), resulted in getting the same slot twice. 
so far I wasn't able to write a test that reproduces it, might be easier if we mock the whole clock. 
So, IMO, this "fixes" the issue but with realizing exactly what caused it to happen. I'd rather we have some test included but I think this should go in anyhow.